tensorflow_addons/seq2seq/beam_search_decoder_test.py

```diff
@@ -530,139 +529,117 @@ def get_probs():
-            expected_seq_length = 3 if tf.executing_eagerly() else None
-            self.assertEqual(
-                _t((batch_size, expected_seq_length, beam_width)),
-                tuple(beam_search_decoder_output.scores.get_shape().as_list()),
-            )
-            self.assertEqual(
-                _t((batch_size, expected_seq_length, beam_width)),
-                tuple(final_outputs.predicted_ids.get_shape().as_list()),
-            )
-
-            self.evaluate(tf.compat.v1.global_variables_initializer())
-            eval_results = self.evaluate(
-                {
-                    "final_outputs": final_outputs,
-                    "final_sequence_lengths": final_sequence_lengths,
-                }
-            )
-
-            max_sequence_length = np.max(eval_results["final_sequence_lengths"])
-
-            # A smoke test
-            self.assertEqual(
-                _t((batch_size, max_sequence_length, beam_width)),
-                eval_results["final_outputs"].beam_search_decoder_output.scores.shape,
-            )
-            self.assertEqual(
-                _t((batch_size, max_sequence_length, beam_width)),
-                eval_results[
-                    "final_outputs"
-                ].beam_search_decoder_output.predicted_ids.shape,
-            )
-
-    def testDynamicDecodeRNNBatchMajorNoAttention(self):
-        self._testDynamicDecodeRNN(time_major=False, has_attention=False)
-
-    def testDynamicDecodeRNNBatchMajorYesAttention(self):
-        self._testDynamicDecodeRNN(time_major=False, has_attention=True)
-
-    def testDynamicDecodeRNNBatchMajorYesAttentionWithAlignmentHistory(self):
-        self._testDynamicDecodeRNN(
-            time_major=False, has_attention=True, with_alignment_history=True
+def _test_dynamic_decode_rnn(time_major, has_attention, with_alignment_history=False):
+    encoder_sequence_length = np.array([3, 2, 3, 1, 1])
+    decoder_sequence_length = np.array([2, 0, 1, 2, 3])
+    batch_size = 5
+    decoder_max_time = 4
+    input_depth = 7
+    cell_depth = 9
+    attention_depth = 6
+    vocab_size = 20
+    end_token = vocab_size - 1
+    start_token = 0
+    embedding_dim = 50
+    max_out = max(decoder_sequence_length)
+    output_layer = tf.keras.layers.Dense(vocab_size, use_bias=True, activation=None)
+    beam_width = 3
+
+    batch_size_tensor = tf.constant(batch_size)
+    embedding = np.random.randn(vocab_size, embedding_dim).astype(np.float32)
+    cell = tf.keras.layers.LSTMCell(cell_depth)
+    initial_state = cell.get_initial_state(batch_size=batch_size, dtype=tf.float32)
+    coverage_penalty_weight = 0.0
+    if has_attention:
+        coverage_penalty_weight = 0.2
+        inputs = tf.compat.v1.placeholder_with_default(
+            np.random.randn(batch_size, decoder_max_time, input_depth).astype(
+                np.float32
+            ),
+            shape=(None, None, input_depth),
```

Comment:
In eager mode, `placeholder_with_default` returns fully defined shapes. So we may need to convert part of this test to use a `tf.function` with an input signature. What do you think?

Reply:
Overall, I'd prefer avoiding `tf.compat.v1` as it's not TF2 idiomatic code. Let me try to remove this part of the diff of this pull request so that we can merge the rest and then I'll open a new pull request for this specific piece of code.

Reply:
I tried to make it work with a `tf.keras.layers.Input` but that didn't go well. I'll drop this PR for now. I may have better ideas in the future.

I believe that we should test with code that looks like user code, that's why I'd prefer using a input layer rather than a tf.function on this one. But if we use tf.function that's not so bad too if we can keep the error checking code out of the tf.function.